### PR TITLE
fix: Can't find app image when swizzling

### PR DESCRIPTION
**Pull request by philipphofmann**
## :scroll: Description

The SDK can't get the image name with the app delegate class for some apps. Therefore, we
use the rootViewController and its subclasses as a fallback.

## :bulb: Motivation and Context

A potential customer complained that they weren't getting any auto-generated transactions. While we were not able to come up with a repro, the customer confirmed that this solution fixes the problem for them.

Fixes GH-1693

## :green_heart: How did you test it?

Asked customer to test with this branch, and it worked.

## :pencil: Checklist

- [x] I reviewed the submitted code
- [x] I added tests to verify the changes
- [x] I updated the docs if needed
- [x] Review from the native team if needed
- [x] No breaking changes

### CODE REVIEW

File: Sources/Sentry/SentryUIViewControllerSwizzling.m [hunk 1]
```diff
@@ -261,10 +286,15 @@ - (void)swizzleRootViewControllerAndDescendant:(UIViewController *)rootViewContr
...
-
+            [SentryLog
+                logWithMessage:@"UIViewControllerSwizziling Calling swizzleRootViewController."
+                      andLevel:kSentryLevelDebug];
             [self swizzleViewControllerSubClass:viewControllerClass];
+
+            // We can't get the image name with the app delegate class for some apps. Therefore, we
+            // use the rootViewController and its subclasses as a fallback.  The following method
+            // ensures we don't swizzle ViewControllers of UIKit.
+            [self swizzleUIViewControllersOfClassesInImageOf:viewControllerClass];
```
Comment (brustolin) [inline Sources/Sentry/SentryUIViewControllerSwizzling.m:297]: m: I know swizzling has duplication protection and usually the root view controller doesn't have many childViewControllers, but, can we improve this somehow? Let's say we have 4 child controllers, so we list all the binary controllers 4 times, filter them and try to swizzle.

Comment (philipphofmann) [reply Sources/Sentry/SentryUIViewControllerSwizzling.m:297]: What about keeping track of the already swizzled images? We could skip then.

Comment (brustolin) [reply Sources/Sentry/SentryUIViewControllerSwizzling.m:297]: Yes, keeping track of swizzled images would work.

**Review by brustolin — APPROVED** [review]
LGTM

**Pull request merged**